packages/amplication-server/src/core/build/utils/previousBuild/previousBuild.ts

```diff
@@ -0,0 +1,24 @@
+import { Build } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+
+export const previousBuild = async (
+  prisma: PrismaService,
+  appId: string,
+  newBuildId: string,
+  buildDate: Date
+): Promise<Build | null> => {
+  const oldBuild = (
+    await prisma.build.findMany({
+      take: 1,
+      skip: 0,
+      where: {
+        appId: appId,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        NOT: { id: newBuildId },
+        createdAt: { lt: buildDate }
+      },
+      orderBy: { createdAt: 'desc' }
+    })
+  ).shift();
```

Comment:
`skip:0` is the default value so you don't have to use it.
New build-id created-at should be equal to build-date,
so newBuild.createdAt < buildDate will always be false.
So the `NOT: { id: newBuildId }` is not mandatory.